doc/users/dflt_style_changes.rst

```diff
@@ -418,100 +462,520 @@ or by setting::
+two ticks visible.
+
+.. plot::
+
+   import matplotlib.pyplot as plt
+   import numpy as np
+
+   from matplotlib.ticker import AutoLocator
+
+   fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(4, 3), tight_layout=True)
+   ax1.set_xlim(0, .1)
+   ax2.set_xlim(0, .1)
+
+   ax1.xaxis.get_major_locator().set_params(nbins=9, steps=[1, 2, 5, 10])
+   ax1.set_title('classic')
+   ax2.set_title('v2.0')
+
+There is no way, other than using ``mpl.style.use('classic')``, to restore the
+previous behavior as the default.  On an axis-by-axis basis you may either
+control the existing locator via: ::
+
+  ax.xaxis.get_major_locator().set_params(nbins=9, steps=[1, 2, 5, 10])
+
+or create a new `~matplotlib.ticker.MaxNLocator`::
+
+  import matplotlib.ticker as mticker
+  ax.set_major_locator(mticker.MaxNLocator(nbins=9, steps=[1, 2, 5, 10])
+
+
+Auto limits
+-----------
+
+The previous auto-scaling behavior was to find 'nice' round numbers
+as view limits that enclosed the data limits, but this could produce
+bad plots if the data happened to fall on a vertical or
+horizontal line near the chosen 'round number' limit.  The new default
+sets the view limits to 5% wider than the data range.
+
+.. plot::
+
+   import matplotlib as mpl
+   import matplotlib.pyplot as plt
+   import numpy
+
+   data = np.zeros(1000)
+   data[0] = 1
+
+   fig = plt.figure(figsize=(6, 3))
+
+   def demo(fig, rc, title, j):
+       with mpl.rc_context(rc=rc):
+           ax = fig.add_subplot(1, 2, j)
+           ax.plot(data)
+           ax.set_title(title)
+
+   demo(fig, {'axes.autolimit_mode': 'round_numbers',
+              'axes.xmargin': 0,
+              'axes.ymargin': 0}, 'classic', 1)
+   demo(fig, {}, 'v2.0', 2)
+
+The size of the padding in the x and y directions is controlled by the
+``'axes.xmargin'`` and ``'axes.ymargin'`` rcParams respectively. Whether
+the view limits should be 'round numbers' is controlled by the
+``'axes.autolimit_mode'`` rcParam.  In the original ``'round_number'`` mode,
+the view limits coincide with ticks.  With the new default value, ``'data'``,
+the outermost ticks will usually be inside the view limits, not at the ends.
+Also see `~matplotlib.axes.Axes.margins`.
+
+For a few `~matplotlib.artist.Artist` classes, margins are undesirable.
+For example, a margin should not be added for a `~matplotlib.image.AxesImage`
+created with `~matplotlib.axes.Axes.imshow`.  To control the application of
+the margins, the `~matplotlib.artist.Artist` class has gained the properties :
+
+    - `~matplotlib.artist.Artist.top_margin`
+    - `~matplotlib.artist.Artist.bottom_margin`
+    - `~matplotlib.artist.Artist.left_margin`
+    - `~matplotlib.artist.Artist.right_margin`
+    - `~matplotlib.artist.Artist.margins`
+
+along with the complimentary ``get_*`` and ``set_*`` methods.  When
+computing the view limits, each `~matplotlib.artist.Artist` is checked.
+If *any* artist does not call for a margin on a given side, the margin
```

Comment:
The logic is the `AND` of all of the `a.*_margin` values, so if a single artist _forbids_ using margins then the margin is not applied.